theano/gof/cmodule.py

```diff
@@ -2277,7 +2283,7 @@ def compile_str(module_name, src_code, location=None,
 
         lib_filename = os.path.join(
             location,
-            '%s.%s' % (module_name, get_lib_extension()))
+            '%s.%s.%s' % (module_name, "pypy-41", get_lib_extension()))
```

Comment:
I would do something like this:
pypy_name = ""
if pypy:
    pypy_name=get_the_pypy_name with regex as talked above + "."
lib_filename = os.path.join(
             location,
            '%s.%s%s' % (module_name, pypy_name, get_lib_extension()))